gap/semigroups/semitrans.gi

```diff
@@ -735,6 +735,47 @@ function(S)
   return t;
 end);
 
+InstallMethod(WreathProduct, "for a monoid and a permutation group",
+[IsTransformationSemigroup, IsPermGroup],
+function(S, G)
+  local maps, gensS, next, reps, n, i, g, x, m;
+  if not IsMonoidAsSemigroup(S) then
+    ErrorNoReturn("S should be a monoid");
+  fi;
+ 
+  m := LargestMovedPoint(G);
+  maps := []; # the final generating set for the wreath product
+
+  gensS := ShallowCopy(GeneratorsOfSemigroup(S));
```

Comment:
We do need ShallowCopy because we later modify the list gensS which would otherwise be immutable as GeneratorsOfSemigroup(S) returns an immutable object.